packages/server-core/src/media/storageprovider/s3.storage.ts

```diff
@@ -76,49 +92,49 @@ export class S3Provider implements StorageProviderInterface {
    * @param fileName Name of file in the storage.
    * @param directoryPath Directory of file in the storage.
    */
-  async doesExist(fileName: string, directoryPath: string): Promise<boolean> {
+  async doesExist(fileName: string, directoryPath: string): Promise<any> {
     // have to use listOBjectsV2 since other object related methods does not check existance of a folder on S3
-    const result = await this.provider
-      .listObjectsV2({
-        Bucket: this.bucket,
-        Prefix: path.join(directoryPath, fileName),
-        MaxKeys: 1
-      })
-      .promise()
-      .then((res) => (res.Contents && res.Contents.length > 0) || false)
-      .catch(() => false)
-
-    return result
+    const result = new ListObjectsV2Command({
+      Bucket: this.bucket,
+      Prefix: path.join(directoryPath, fileName),
+      MaxKeys: 1
+    })
+    try {
+      const response = await this.provider.send(result)
+      ;(response.Contents && response.Contents.length > 0) || false
+    } catch {
+      false
+    }
   }
-
   /**
    * Check if an object is directory or not.
    * @param fileName Name of file in the storage.
    * @param directoryPath Directory of file in the storage.
    */
-  async isDirectory(fileName: string, directoryPath: string): Promise<boolean> {
+  async isDirectory(fileName: string, directoryPath: string): Promise<any> {
```

Comment:
Same deal here as doesExist - the returned value type should be Promise<boolean>, and line 124 and 126 should be returning those values.